File: bindings/ruby/examples/slots.rb
#!/usr/bin/ruby

require 'storage'


environment = Storage::Environment.new(true)

storage = Storage::Storage.new(environment)
storage.probe()

probed = storage.probed()

partitionable = Storage::Partitionable::find_by_name(probed, "/dev/sdd")

partition_table = partitionable.partition_table()

partition_slots = partition_table.unused_partition_slots()

partition_slots.each() do |partition_slot|

  puts "#{partition_slot}"

end

File: bindings/python/examples/slots.py
#!/usr/bin/python3

from storage import Environment, Storage, Partitionable


environment = Environment(True)

storage = Storage(environment)
storage.probe()

probed = storage.get_probed()

partitionable = Partitionable.find_by_name(probed, "/dev/sdd")

partition_table = partitionable.get_partition_table()

partition_slots = partition_table.get_unused_partition_slots()

for partition_slot in partition_slots:

  print("%s" % partition_slot)

